datadog/provider.go

```diff
@@ -470,12 +470,14 @@ func tagDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) erro
 		kv := strings.Split(tag.(string), ":")
```

Comment:
You could probably use `strings.Cut` (or even `strings.SplitN`) and avoid having to `strings.Join` afterwards. This would probably allow to simplify the whole check logic!

Reply:
Like so:
```go
    tagStr := tag.(string)
    if tagStr == "" {
        return fmt.Errorf("invalid tag: '%s'", tag)
    }
    key, value, found := strings.Cut(tagStr, ":")
    if !found {
        value = ""
    }
    tags[key] = value
```

Reply:
Good point - I always forget about this function.

Reply:
Done - I've even removed the special case for empty tags because it does not break the code.